src/circuit/gadget/poseidon/pow5.rs

```diff
@@ -705,9 +716,11 @@ mod tests {
                         region.constrain_equal(final_state[i].var, var)
                     };
 
-                    final_state_word(0)?;
-                    final_state_word(1)?;
-                    final_state_word(2)
+                    for i in 0..(WIDTH - 1) {
+                        final_state_word(i)?;
+                    }
+
+                    final_state_word(WIDTH - 1)
```

Comment:
I think this is slightly cleaner and more resilient:
```suggestion
                    for i in 0..WIDTH {
                        final_state_word(i)?;
                    }

                    Ok(())
```